Gem Detonator (5) and Gem Dependant fix

senor-huevo opened the pull request:
Fixes Gem Detonator (5) to count itself, as requested by SylveonM45T3R.

Adds customization to Gem Dep to work how it does in base game.

adds new bool to the JSON called: `vanilla_gem_dep`

It's only used for disabling the death trigger of Gem Dep to make it able to be checked at the end or start of turn.

Mouthless-Stoat reviewed (dismissed):
I don't know how the new sigil system work yet but it seem like you only implemented the gem dep fix for just the old system and not the newer one?

Mouthless-Stoat reviewed (changes requested):
I don't know how the new sigil system work yet but it seem like you only implemented the gem dep fix for just the old system and not the newer one?

whiterobot10 commented:
> I don't know how the new sigil system work yet but it seem like you only implemented the gem dep fix for just the old system and not the newer one?

The new system runs in series with the old system and only does things the old system was incapable of doing.

senor-huevo commented:
> The new system runs in series with the old system and only does things the old system was incapable of doing.

I assume Stoat was referring to the great renaming

senor-huevo committed:
Gem Dep and Gem Det (5) fix

updated to be in the Great Renaming and with Stoat's requests for Gem Dep

senor-huevo committed:
Unable to Sac bug

I was unable to sac creatures because of some of the variables were not renamed to be in line with the great renaming.
Specifically sacVictims not being sac_victims

senor-huevo committed:
Fixing the mistake I made

Didn't realize that erasing Pharoah.card did absolutely nothing, so I rectified my error.

Mouthless-Stoat commented on scripts/CardFight.gd:
```diff
@@ -1206,8 +1206,11 @@ func start_turn():
 	for pharoah in gold_sarcophagus:
 		if pharoah.turnsleft <= 0:
-			draw_card(pharoah.card)
-			gold_sarcophagus.erase(pharoah.card)
+			if pharoah.turnsleft < 0: 
+				gold_sarcophagus.erase(pharoah)
+			else:
+				draw_card(pharoah.card)
+				pharoah.turnsleft -= 1
 		else:
 			pharoah.turnsleft -= 1
```
The new check that you added here is redundant, The check for `if pharoah.turnsleft < 0` is in another check for the same thing. You should eliminate the outer conditional and just use the inner one.

Mouthless-Stoat commented on scripts/classes/sigils/Gem Dependant.gd:
```diff
@@ -1,17 +1,21 @@
 	# attached_card_summoned represents the card bearing the sigil being summoned
-	if (event == "card_summoned" and params[0] == card) or (event == "card_perished" and params[0] != card):
+	# Left side is checking when the card is played, Right side is checking when anything perishes
+	if (event == "card_summoned" and params[0] == card and not card.in_hand) or (event == "card_perished" and params[0] != card and not card.in_hand and CardInfo.all_data.vanilla_gem_dep != true):
```
This may benefit from a small refactoring by extracting each of the condition into a small variable or a comment to better explain what each of the condition is.

senor-huevo replied:
<img width="1096" height="204" alt="Screen Shot 2025-07-24 at 1 18 21 PM" src="https://github.com/user-attachments/assets/eb0c8682-3258-4922-aa11-1a809e9d1f15" />
I moved up the check for if the card is not in hand since both sides had the check and added a backslash to make it more readable instead of it being a long contiguous statement.

Should do I anything else?

Mouthless-Stoat replied:
Might be beneficial to give them name like
```gdscript
summoned = event == "card_summoned" and params[0] == card
gem_perished = event == "card_perished" and params[0] != card and not card.in_hand and CardInfo.all_data.vanilla_gem_dep != true
```
just make it clearer what each part of the clause mean

Mouthless-Stoat reviewed (changes requested):
Just a few more small code fix and style and this should be good to merge.
